flink-tests/src/test/java/org/apache/flink/test/recovery/ProcessFailureCancelingITCase.java

```diff
@@ -331,4 +337,17 @@ private void printProcessLog(String processName, String log) {
 		System.out.println("		END SPAWNED PROCESS LOG");
 		System.out.println("-----------------------------------------");
 	}
+
+	private int findFreePort(int startPort, int range) {
+		for (int p = startPort; p < startPort + range; ++p) {
+			try {
+				new ServerSocket(p).close();
```

Comment:
this is not a safe way to determine free ports, as they can still be claimed by another process after this method returns and before we claim it.